src/common/expression/ParameterExpression.cpp

```diff
@@ -0,0 +1,30 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "common/expression/ParameterExpression.h"
+
+#include "common/expression/ExprVisitor.h"
+
+namespace nebula {
+
+const Value& ParameterExpression::eval(ExpressionContext& ectx) { return ectx.getVar(name_); }
+
+std::string ParameterExpression::toString() const { return name_; }
+
+bool ParameterExpression::operator==(const Expression& rhs) const {
+  return kind_ == rhs.kind() && name_ == rhs.toString();
+}
+
+void ParameterExpression::writeTo(Encoder& encoder) const {
+  encoder << kind_;
+  encoder << name_;
+}
```

Comment:
Will this expression be passed to the storage layer?

Reply:
Not yet.ParameterExpression relies on graph::ExecutionContext::valueMap_.